src/test/java/nextstep/subway/unit/LineServiceTest.java

```diff
@@ -23,11 +31,111 @@ public class LineServiceTest {
     void addSection() {
         // given
         // stationRepository와 lineRepository를 활용하여 초기값 셋팅
+        Line line = lineRepository.save(new Line("8호선", "bg-pink-500"));
+        Station upStation = stationRepository.save(new Station("암사역"));
+        Station downStation = stationRepository.save(new Station("모란역"));
 
         // when
         // lineService.addSection 호출
+        lineService.addSection(line.getId(), new SectionRequest(upStation.getId(), downStation.getId(), 20));
 
         // then
         // line.getSections 메서드를 통해 검증
+        assertAll(() -> {
+            assertThat(line.getSections().get(0).getUpStation()).isEqualTo(upStation);
+            assertThat(line.getSections().get(0).getDownStation()).isEqualTo(downStation);
+        });
```

Comment:
```suggestion
        assertAll(() -> {
            assertThat(line.getSections().get(0).getUpStation()).isEqualTo(upStation);
            assertThat(line.getSections().get(0).getDownStation()).isEqualTo(new Station("암사역"));
        });
```

단위 테스트에 대한 검증을 할 때 이미 만들어진 객체로 검증을 해도 되려나요? 아니면, 새로운 객체를 만들어서 검증을 해야 하려나요?

 지난번 미션에서는 항상 새로운 객체를 만들어서 검증을 했는데, 요번에는 이미 만들어진 객체로 검증을 하면 재사용성도 좋고, 값에 대한 검증이 확실할 것 같다는 생각이 들었습니다.

리뷰어님은 단위 테스트에 대한 검증을 어떻게 하고 있으신가요? 궁금합니다! 😊

Reply:
오 좋은 질문 감사합니다 :)

저는 그냥 getter를 통해 필드 값을 비교하는 편이긴 합니다! 👀 
DTO나 VO 같이 복잡한 객체를 비교할 때는 새로운 객체를 생성해서 비교하기도 하구요..!

그리고 저는 이런 Service layer 테스트 시 객체를 직접적으로 재사용 하는 방식은 되도록 안하려고 하는데요.
실제로 우리가 코드를 사용하는 방식과 위배된다고 생각해서입니다.

좀 더 풀어서 설명드리면 우리가 `LineService#addSection`을 통해 노선에 구간을 추가하고
추후에 노선의 구간을 조회할 때는 `DB`에서 다시 조회하지 메모리에 있는 객체를 가져다가 쓰진 않을 거에요.
그런 의미로 저는 지금과 같은 경우는 Line 객체도 재사용을 하기 보다는, then 절에서 lineRepository를 통해 노선을 조회한 후에 assert를 하는 편입니다 :)

Reply:
오! 좋은 의견 감사합니다!!!